src/pages/Trade/components/PriceChart/components/Chart/ChartContext.tsx

```diff
@@ -0,0 +1,167 @@
+import Button from 'components/Button';
+import { Positions } from 'enums/options';
+import { createChart, ColorType, IChartApi } from 'lightweight-charts';
+import React, { createContext, useEffect, useRef, useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { ThemeInterface } from 'types/ui';
+import { CandlestickComponent } from './CandlestickSeries/CandlestickComponent';
+import { AreaSeriesComponent } from './AreaSeries/AreaSerierComponent';
+import { UserPositionAreaSeries } from './UserSeries/UserSeriesComponent';
+
+export const ChartContext = createContext<IChartApi | null>(null);
+
+type ChartContextProps = {
+    children: React.ReactNode;
+    chart: IChartApi | null;
+};
+
+type ChartProps = {
+    data: any;
+    position: Positions | undefined;
+    asset: string;
+    isSpeedMarkets: boolean;
+    selectedPrice?: number;
+    selectedDate?: number;
+};
+
+const ChartProvider: React.FC<ChartContextProps> = ({ children, chart }) => (
+    <ChartContext.Provider value={chart}>{children}</ChartContext.Provider>
+);
+
+export const ChartComponent: React.FC<ChartProps> = ({
+    data,
+    position,
+    asset,
+    isSpeedMarkets,
+    selectedPrice,
+    selectedDate,
+}) => {
+    const theme: ThemeInterface = useTheme();
+    const chartContainerRef = useRef<HTMLDivElement>(null);
+    const [chart, setChart] = useState<IChartApi | undefined>();
+    const [areaData, setAreaData] = useState();
+
+    useEffect(() => {
+        if (selectedPrice && selectedDate && position && data) {
+            const lineDataSelected = data.map((datapoint: any) => ({
+                time: datapoint.time,
+                value: selectedPrice,
+            }));
+            const deltaTime = data[1].time - data[0].time;
+            while (lineDataSelected[lineDataSelected.length - 1].time + deltaTime < Math.floor(selectedDate / 1000)) {
```

Comment:
`Math.floor(selectedDate / 1000)`
Just as info date-fns has function `millisecondsToSeconds` and similar which are more readable.